test/forcing/ForcingsEngineDataProvider_Test.cpp

```diff
@@ -0,0 +1,183 @@
+        //        See issue #748 for more info, keeping this temporarily
+        //        until ESMF implements a way to keep MPI initialized on
+        //        ESMF finalization through ESMPy.
+        // gil_->getModule("atexit").attr("_run_exitfuncs")();
+
+        data_access::ForcingsEngine::finalize_all();
+        gil_.reset();
+
+        #if NGEN_WITH_MPI
+        // MPI_Finalized(&mpi.finalized);
+        // if (mpi.finalized == 0) {
+        PMPI_Finalize();
+        // }
+        #endif
+    }    
+};
+
+std::shared_ptr<utils::ngenPy::InterpreterUtil> ForcingsEngineDataProviderTest::gil_{};
+data_access::ForcingsEngineDataProvider ForcingsEngineDataProviderTest::provider{};
+mpi_info ForcingsEngineDataProviderTest::mpi{};
+
+// Tests that the forcings engine data provider correctly indexes epochs
+// to unitless indices along its given temporal domain.
+TEST_F(ForcingsEngineDataProviderTest, Timestepping)
+{
+    EXPECT_EQ(this->provider.get_ts_index_for_time(this->params.simulation_start_t), 0);
+    EXPECT_EQ(this->provider.get_ts_index_for_time(this->params.simulation_end_t), 5);
+}
+
+// Tests that the forcings engine correctly initializes and performs
+// a call to `get_values`.
+TEST_F(ForcingsEngineDataProviderTest, Initialization)
+{
+    std::cout << "Getting variable names\n";
+    const auto outputs_test = this->provider.get_available_variable_names();
+    std::cout << "Got variable names\n";
+    constexpr std::array<const char*, 8> expected_variables = {
+        "U2D_ELEMENT",
+        "V2D_ELEMENT",
+        "LWDOWN_ELEMENT",
+        "SWDOWN_ELEMENT",
+        "T2D_ELEMENT",
+        "Q2D_ELEMENT",
+        "PSFC_ELEMENT",
+        "RAINRATE_ELEMENT"
+    };
+    
+    for (decltype(auto) output : boost::combine(outputs_test, expected_variables)) {
+        EXPECT_EQ(output.get<0>(), output.get<1>());
+    }
+
+    std::cout << "Creating selector\n";
+    const auto selector = CatchmentAggrDataSelector{
+        "cat-1015786",
+        "RAINRATE",
+        this->provider.get_data_start_time(), // start time
+        static_cast<long>(this->provider.get_ts_index_for_time(this->params.simulation_start_t) + (3600 * 2)), // duration
+        "seconds"
+    };
+
+    std::cout << "Getting values\n";
+    const auto result = this->provider.get_values(
+        selector,
+        data_access::ReSampleMethod::SUM
+    );
+
+    for (const auto& r : result) {
+        EXPECT_NEAR(r, 5.51193e-07, 1e6);
+    }
+}
+
+TEST_F(ForcingsEngineDataProviderTest, Lookback)
+{
+    // Both selectors encompass the entire temporal domain,
+    // and since they're AOIs, they should have different values.
+    // This means, to test lookback, we can evaluate one selector,
+    // then attempt to use the second selector. If the values match
+    // then lookback did not work correctly. Otherwise, lookback worked.
+    //
+    // Note that this assumes the variable values are different for the SAME variable.
```

Comment:
FYI, this is slightly outdated I believe. I initially was testing the difference in value between catchments, but just below I'm testing the difference between variables in the same catchment.

Reply:
To clarify, this test needs to be reworked since it's not testing anything meaningful I think.